src/colorizer/ColorizeCanvas.ts

```diff
@@ -487,20 +496,24 @@ export default class ColorizeCanvas {
     // TODO: Optimize memory by using a true boolean array? Bit-level manipulation to fit it within Uint8Array?
     // TODO: If optimizing, use fuse operation via shader.
     const inRangeIds = new Uint8Array(this.dataset.numObjects);
-    inRangeIds.fill(1);
 
-    for (const threshold of thresholds) {
-      const featureData = this.dataset.getFeatureData(threshold.featureName);
-      // Ignore thresholds with features that don't exist in this dataset or whose units don't match
-      if (!featureData || featureData.units !== threshold.units) {
-        continue;
-      }
-      for (let i = 0, n = inRangeIds.length; i < n; i++) {
-        if (inRangeIds[i] === 1 && (featureData.data[i] < threshold.min || featureData.data[i] > threshold.max)) {
-          inRangeIds[i] = 0;
+    // Ignore thresholds with features that don't exist in this dataset or whose units don't match
+    const validThresholds = thresholds.filter((threshold) => {
+      const featureData = this.dataset?.tryGetFeatureData(threshold.featureName);
+      return featureData && featureData.units === threshold.units;
+    });
+
+    inRangeIds.forEach((_value, id, arr) => {
```

Comment:
Replace with integer for loop

Reply:
If you consider this a performance sensitive part of the code, we have measured something like at least 30% perf improvement by using the old style integer for loops.
Consider changing the innermost loop to do the same if possible.